tree/ntuple/v7/src/RNTupleProcessor.cxx

```diff
@@ -17,13 +17,72 @@
 
 #include <ROOT/RFieldBase.hxx>
 
+namespace {
+using ROOT::Experimental::RNTupleSourceSpec;
+void EnsureUniqueNTupleNames(std::vector<RNTupleSourceSpec> ntuples)
+{
+   std::sort(ntuples.begin(), ntuples.end(),
+             [](const RNTupleSourceSpec &left, const RNTupleSourceSpec &right) { return left.fName < right.fName; });
+   auto newEnd =
+      std::unique(ntuples.begin(), ntuples.end(), [](const RNTupleSourceSpec &left, const RNTupleSourceSpec &right) {
+         return left.fName == right.fName;
+      });
+   if (newEnd != ntuples.end()) {
+      throw ROOT::Experimental::RException(R__FAIL("horizontal joining of RNTuples with the same name is not allowed"));
```

Comment:
I see the need for this exception now, because otherwise one could not disambiguate when calling field `main_name.x` and field `auxiliary_name.x` in case `auxiliary_name == main_name`. In TTree this situation would be supported by giving the auxiliary dataset an alias. Something to keep in mind, maybe add a comment for the future.

Reply:
Yes, this is something I realized as well. I have a mental TODO but will add explicitly add it as well :-).